update-versions.py

```diff
@@ -1,50 +1,90 @@
 import argparse
 import collections
 import functools
-import github
 import json
 import os
 import pathlib
-import semver
 import subprocess
+from dataclasses import dataclass
+from pathlib import Path
+from typing import (
```

Comment:
@pedorich-n this is a great improvement! Thank you for adding type-hints throughout the script.